modules/programs/firefox/mkFirefoxModule.nix

```diff
@@ -666,37 +662,114 @@ in {
               for more information.
             '';
           };
-
           extensions = mkOption {
-            type = types.listOf types.package;
-            default = [ ];
-            example = literalExpression ''
-              with pkgs.nur.repos.rycee.firefox-addons; [
-                privacy-badger
-              ]
-            '';
-            description = ''
-              List of ${name} add-on packages to install for this profile.
-              Some pre-packaged add-ons are accessible from the
-              [Nix User Repository](https://github.com/nix-community/NUR).
-              Once you have NUR installed run
+            type = types.coercedTo (types.listOf types.package) (packages: {
+              packages = mkIf (builtins.length packages > 0) (warn ''
+                In order to support declarative extension configuration,
+                extension installation has been moved from
+                programs.firefox.profiles.<profile>.extensions
```

Comment:
Any instances of `programs.firefox` in the documentation should be replaced with `moduleName` since this file meant to be re-usable in firefox forks.